manifests/e/EclipseFoundation/AdoptiumOpenJDK/17/17.0.0.35/EclipseFoundation.AdoptiumOpenJDK.17.locale.en-US.yaml

```
@@ -0,0 +1,19 @@
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultlocale.1.0.0.schema.json
+PackageIdentifier: EclipseFoundation.AdoptiumOpenJDK.17
+PackageVersion: 17.0.0.35
+PackageLocale: en-US
+LicenseUrl: https://adoptium.net/about.html
+Moniker: adoptiumopenjdk17
+Tags:
+- openjdk
+- java
+- jvm
+- jdk
+PackageUrl: https://adoptium.net
+Publisher: Eclipse Foundation
+PackageName: Eclipse Adoptium Open JDK 17
```

This is not a required change, but this would MAYBE help with the matching for older versions which is not in the WinGet Repo. I say maybe because the PackageName is actually `Eclipse Temurin JDK with Hotspot 17+35 (x64)`
```suggestion
PackageName: Eclipse Temurin JDK with Hotspot 17
```

---

Should we also change the product name? (rename the files)

---

No, the `PackageIdentifier` is independent and just defines the Folder/Package path and file names.

> @OfficialEsco thanks for your comments! On top of that I had a question. Because of the new manifest format, the major version in in the package name. Wouldn't this prevent upgrades? (from 16 to 17)

When the `PackageIdentifier` is different its a new Package, `EclipseFoundation.AdoptiumOpenJDK.16` and `EclipseFoundation.AdoptiumOpenJDK.17` are 2 different packages and will only see new updates when a new version is added to the same `PackageIdentifier` aka `EclipseFoundation.AdoptiumOpenJDK.17` version `17.0.0.36`

Winget-Cli looks at the ARP table for packages matching the `ProductCode`, `Publisher` and `PackageName`, winget-cli also removes architecture and version from the search and there is a wildcard behind the search (`Eclipse Temurin JDK with Hotspot 17 *`)

So technically the `PackageName` i suggested is wrong because it should be `Eclipse Temurin JDK with Hotspot 17+35`

---

So should we change the package identifier across both major versions to make upgrades possible? Should I also change the folder name while I do that?

---

Nuuu, because this is Java and you can have multiple major version installed they should not update to a different Major version

---

right, this makes sense, I didn't think of this scenarios (I always go with latest on my side).
Are we ok with this?

```suggestion
PackageName: Eclipse Adoptium Temurin JDK with Hotspot 17+35
```

---

It depends what you use JDK for, some software are not compatible, stable or optimized for newer major versions of JDK.

And no, since the full PackageName is `Eclipse Temurin JDK with Hotspot 17+35 (x64)`
If we get support for wildcard later we can do `Eclipse Temurin JDK with Hotspot 17*` since atm the wildcard works like `Eclipse Temurin JDK with Hotspot 17 *`

`Eclipse Temurin JDK with Hotspot 17+35` a useless `PackageName` since it specifies the minor version in the `PackageName`

`Eclipse Adoptium Temurin JDK with Hotspot 17` will always fail to find a match because the real `PackageName` does not have `Adoptium` at the start

---

sorry, I'm confused now. What do you want me to change it to?

---

Eclipse Temurin JDK with Hotspot 17

---

```suggestion
PackageName: Eclipse Temurin JDK with Hotspot 17
```